src/Lights/Shadows/cascadedShadowGenerator.ts

```diff
@@ -0,0 +1,1708 @@
+        if (!this._scene.activeCamera) {
+            return;
+        }
+
+        const prevSplitDist = this._cascades[cascadeIndex].prevBreakDistance,
+              splitDist = this._cascades[cascadeIndex].breakDistance;
+
+        this._scene.activeCamera.getViewMatrix(); // make sure the transformation matrix we get when calling 'getTransformationMatrix()' is calculated with an up to date view matrix
+
+        const invViewProj = Matrix.Invert(this._scene.activeCamera.getTransformationMatrix());
+        for (let cornerIndex = 0; cornerIndex < CascadedShadowGenerator.frustumCornersNDCSpace.length; ++cornerIndex) {
+            Vector3.TransformCoordinatesToRef(CascadedShadowGenerator.frustumCornersNDCSpace[cornerIndex], invViewProj, this._frustumCornersWorldSpace[cascadeIndex][cornerIndex]);
+        }
+
+        // Get the corners of the current cascade slice of the view frustum
+        for (let cornerIndex = 0; cornerIndex < CascadedShadowGenerator.frustumCornersNDCSpace.length / 2; ++cornerIndex) {
+            tmpv1.copyFrom(this._frustumCornersWorldSpace[cascadeIndex][cornerIndex + 4]).subtractInPlace(this._frustumCornersWorldSpace[cascadeIndex][cornerIndex]);
+            tmpv2.copyFrom(tmpv1).scaleInPlace(prevSplitDist); // near corner ray
+            tmpv1.scaleInPlace(splitDist); // far corner ray
+
+            tmpv1.addInPlace(this._frustumCornersWorldSpace[cascadeIndex][cornerIndex]);
+
+            this._frustumCornersWorldSpace[cascadeIndex][cornerIndex + 4].copyFrom(tmpv1);
+            this._frustumCornersWorldSpace[cascadeIndex][cornerIndex].addInPlace(tmpv2);
+        }
+    }
+
+    private _computeCascadeFrustum(cascadeIndex: number): void {
+        this._cascadeMinExtents[cascadeIndex].copyFromFloats(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE);
+        this._cascadeMaxExtents[cascadeIndex].copyFromFloats(Number.MIN_VALUE, Number.MIN_VALUE, Number.MIN_VALUE);
+        this._frustumCenter[cascadeIndex].copyFromFloats(0, 0, 0);
+
+        const camera = this._scene.activeCamera;
+
+        if (!camera) {
+            return;
+        }
+
+        // Calculate the centroid of the view frustum slice
+        for (let cornerIndex = 0; cornerIndex < this._frustumCornersWorldSpace[cascadeIndex].length; ++cornerIndex) {
+            this._frustumCenter[cascadeIndex].addInPlace(this._frustumCornersWorldSpace[cascadeIndex][cornerIndex]);
+        }
+
+        this._frustumCenter[cascadeIndex].scaleInPlace(1 / this._frustumCornersWorldSpace[cascadeIndex].length);
+
+        if (this.stabilizeCascades) {
+            // Calculate the radius of a bounding sphere surrounding the frustum corners
+            let sphereRadius = 0;
+            for (let cornerIndex = 0; cornerIndex < this._frustumCornersWorldSpace[cascadeIndex].length; ++cornerIndex) {
+                const dist = this._frustumCornersWorldSpace[cascadeIndex][cornerIndex].subtract(this._frustumCenter[cascadeIndex]).length();

```

Comment:
Peut on changer le substract par du in place ou toRef ?

Reply:
Ok.